docs/how-to/hip_runtime_api/memory_management/virtual_memory.rst

```diff
@@ -37,53 +37,90 @@ is separated into multiple steps using the :cpp:func:`hipMemCreate`,
 :cpp:func:`hipMemSetAccess` functions. This guide explains what these functions
 do and how you can use them for virtual memory management.
 
+.. _vmm_support:
+
+Virtual memory management support
+---------------------------------
+
+The first step is to check if the targeted device or GPU supports virtual memory management.
+Use the :cpp:func:`hipDeviceGetAttribute` function to get the 
+``hipDeviceAttributeVirtualMemoryManagementSupported`` attribute for a specific GPU, as shown in the following example.
+
+.. code-block:: cpp
+
+    int vmm = 0, currentDev = 0;
+    hipDeviceGetAttribute(
+        &vmm, hipDeviceAttributeVirtualMemoryManagementSupported, currentDev
+    );
+
+    if (vmm == 0) {
+        std::cout << "GPU " << currentDev << " doesn't support virtual memory management." << std::endl;
+    } else {
+        std::cout << "GPU " << currentDev << " support virtual memory management." << std::endl;
+    }
+
+.. _allocate_physical_memory:
+
 Allocate physical memory
---------------------------------------------------------------------------------
+------------------------
 
-The first step is to allocate the physical memory itself with the
+The next step is to allocate the physical memory using the
 :cpp:func:`hipMemCreate` function. This function accepts the size of the buffer,
 an ``unsigned long long`` variable for the flags, and a
 :cpp:struct:`hipMemAllocationProp` variable. :cpp:struct:`hipMemAllocationProp`
 contains the properties of the memory to be allocated, such as where the memory
 is physically located and what kind of shareable handles are available. If the
 allocation is successful, the function returns a value of
 :cpp:enumerator:`hipSuccess`, with :cpp:type:`hipMemGenericAllocationHandle_t`
-representing a valid physical memory allocation. The allocated memory size must
-be aligned with the granularity appropriate for the properties of the
-allocation. You can use the :cpp:func:`hipMemGetAllocationGranularity` function
-to determine the correct granularity.
+representing a valid physical memory allocation.
+
+The allocated memory size must be aligned with the granularity appropriate for
+the properties of the allocation. If the allocation size is not in alignment (not
+dividable) with the minimum granularity of the allocation, the
+:cpp:func:`hipMemCreate` function returns an out of memory error. You can use the
+:cpp:func:`hipMemGetAllocationGranularity` function to determine the correct
+granularity, which is affected by the targeted device hardware and type of
+memory allocation.
```

Comment:
```suggestion
The allocated memory must be aligned with the appropriate granularity. The
granularity can be queried with :cpp:func:`hipMemGetAllocationGranularity`;
the target device hardware and type of memory allocation effect the returned
value. If the allocation size is not in alignment (i.e. not cleanly divisible) with
the minimum granularity value, :cpp:func:`hipMemCreate` returns an
out-of-memory error.
```
This paragraph is quite awkward, and difficult to follow. I have provided a suggested re-phrasing, feel free to use or re-write as you see fit.

Reply:
Yes, the tech writer labeled this paragraph, but I did not find a good replacement. We try to avoid the `i.e`. and `;`. That's why I made slight changes to your suggestion. Please check the outcome.

Reply:
Thanks, looks good